### PR TITLE
Added a YAML linter.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Following https://github.com/opensearch-project/opensearch-build/pull/1480 added a YAML linter and fixer.

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
+        types: [python]
+      - id: flake8
+        stages: [commit]
+        name: flake8
+        entry: bash -c 'pipenv run flake8 .'
+        language: system
+      - id: mypy
+        stages: [commit]
+        name: mypy
+        entry: bash -c 'pipenv run mypy src/build_workflow tests/tests_build_workflow
```
Comment (peternied) [inline .pre-commit-config.yaml:22]: Looks like `line-length` is disabled in yamllint, which is my preference.  However, it seems like ymlfix was applied with this setting enabled.  Can we synchronize the rules they are applied between them?

**Review by peternied — COMMENTED** [review]
Great addition to the codebase, thanks!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dblock) [reply .pre-commit-config.yaml]: Unfortunately yaml-fix doesn't support line length configuration (https://github.com/lyz-code/yamlfix/issues/117) currently.

I manually undid the 2-3 instances where the fixer CR/LFed those lines.

Comment (peternied) [reply .pre-commit-config.yaml]: Maybe we enable the line-length rule so we are consistent, WDYT?

**Review by peternied — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dblock) [reply .pre-commit-config.yaml]: Nan, we actually have a similar problem in Python when we recommend using `black` that auto-formats differently, but it hasn't been a concern.

**Review by tianleh — APPROVED** [review]

**Pull request merged**